CSharpMath/Structures/Trie.cs

```diff
@@ -0,0 +1,216 @@
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace CSharpMath {
+  partial class Extensions {
+    public static void ZipWith<T>(this ReadOnlyMemory<T> @this, ReadOnlyMemory<T> other,
```

Comment:
Needs documentation

Reply:
Added

Reply:
This method is to do with separating numbers and characters in some way?

Reply:
The code does not filter on numbers and characters. `commonHead` is the "common head" of both strings, `thisRest` is the rest of `@this`, `otherRest` is the rest of `other`. The names are self-explanatory enough to not add `<param>` documentation.